presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -8173,4 +8185,49 @@ public void testAssignUniqueId()
                         "WHERE a = 1)",
                 "VALUES 3008750");
     }
+
+    @Test
+    public void testAggregationPushedBelowOuterJoin()
+    {
+        assertQuery(
+                "SELECT * " +
+                        "FROM nation n1 " +
+                        "WHERE (n1.nationkey > ( " +
+                        "SELECT avg(nationkey) " +
+                        "FROM nation n2 " +
+                        "WHERE n1.regionkey=n2.regionkey))"
+        );
+        assertQuery(
+                "SELECT max(name), min(name), count(nationkey) + 1, count(DISTINCT nationkey) " +
```

Comment:
This never goes through the rewrite because there's a mark distinct node between the aggregation and the join.  I'll remove the distinct from the test, but wanted to note this first.